crates/nu-parser/src/parser.rs

```diff
@@ -795,6 +795,9 @@ fn parse_oneof(
             true => parse_multispan_value(working_set, spans, spans_idx, shape),
             false => parse_value(working_set, spans[*spans_idx], shape),
         };
+        if let SyntaxShape::RowCondition = shape {
+            return value;
+        }
```

Comment:
Basically this is only for more sensible help message of `where` command, right?

Would it be better if we move the special check to document generation?

Reply:
How about changing this line:

https://github.com/nushell/nushell/blob/1ba4fe0aac4dba9c9602f0f891bcc8b214f8cbe8/crates/nu-protocol/src/syntax_shape.rs#L242